src/cosalib/aliyun.py

```diff
@@ -82,11 +89,70 @@ def aliyun_run_ore_replicate(build, args):
             } for region, val in ore_data.items()
         ])
 
+    # short circuit the rest of this: if we haven't uploaded to all the specified
+    # regions, we shouldn't be continuing to try to make them public
+    if upload_failed_in_region is not None:
+        raise Exception(f"Upload failed in {upload_failed_in_region} region")
+
     build.meta['aliyun'] = aliyun_img_data
     build.meta_write()
 
-    if upload_failed_in_region is not None:
-        raise Exception(f"Upload failed in {upload_failed_in_region} region")
+    # delay marking images public to allow the cloud to register the newly
+    # created images
```

Comment:
Not sure I fully understand this comment.. What is "delayed" here?

Reply:
I originally had a version of the code that did the `copy-image` followed by `make-public`, which was unreliable as the cloud sometimes had not registered the new image ID in their system.

Delaying the `make-public` operation until after all the `copy-image` operations gives the cloud time to register all the image IDs.  It's not bulletproof, but has been more reliable in my testing.

Reply:
What trips me up is the comment because we're not delaying anything in the code immediately following the comment. After your description here maybe we can make it better.

```
Now that we've ran the image copy in all regions,
let's go through and make them public.
```

Then above the for loop above we can say:

```
Copy the image to all regions. We'll then go mark
them public afterwards since it takes some time
for each image to show up in each region.
```